### PR TITLE
SDK update - types, lerna, minor optimizations in the SDK

### PR DESCRIPTION
**Pull request by danprinz**
Make lerna use separate node_modules for each sub-package
Remove yarn in favor of npm
Update README
[CLI] Allow user to set the amount (with default)
[SDK] get number of retries as a parameter
[SDK] use storage to save redundant computations of tx signatures (and make it unique for every transaction)
[SDK] take tx gas as a parameter (with default)
[SERVER] - correct product reference id(s) to be uuid 4

### CODE REVIEW

**Review by nicolasbrugneaux — COMMENTED** [review]
I will look more thoroughly at the actual code changes ASAP, but I wanted to get first some information regarding the swap to npm from yarn. Could you elaborate the reasoning behind that change?

**Comment by danprinz** [comment]
> I will look more thoroughly at the actual code changes ASAP, but I wanted to get first some information regarding the swap to npm from yarn. Could you elaborate the reasoning behind that change?

We had some issues when installing some of the packages in applications that don't use `yarn`. 
I see this troubles more of you than I've expected, so I reverted it back to use `yarn`. We will deal with `npm` on the other side.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,8 +39,8 @@ jobs:
...
         with:
           path: "**/node_modules"
-          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/package-lock.json') }}
       - name: Install dependencies
-        run: yarn install
+        run: npm install
       - name: Lint
-        run: yarn lint
+        run: npm run lint
```
Comment (nicolasbrugneaux) [inline .github/workflows/ci.yml:46]: let's revert that too

File: packages/sdk/src/handlers/contract-kit.ts [hunk 2]
```diff
@@ -54,22 +88,33 @@ export class ContractKitTransactionHandler implements ChainHandler {
...
+    gasPriceMinimum: BigNumber,
+    txo: CeloTxObject<unknown>
+  ) {
+    const nonce = this.lastNonce ? this.lastNonce++ : await this.getNonce();
+
+    return {
       to: stable.address,
       from: this.blockchainAddress,
-      gas: 100_000,
+      gas: this.gas,
```
Comment (nicolasbrugneaux) [inline packages/sdk/src/handlers/contract-kit.ts:109]: While I know this was there before the refactor, maybe we could use an estimation instead?

**Review by nicolasbrugneaux — COMMENTED** [review]
overall it looks good, let's just clean up the package-locks and the couple places where yarn is still replaced

File: .github/workflows/ci.yml (hunk 1 above)
Comment (danprinz) [reply .github/workflows/ci.yml:46]: reverted

File: packages/sdk/src/handlers/contract-kit.ts (hunk 2 above)
Comment (danprinz) [reply packages/sdk/src/handlers/contract-kit.ts:109]: We might use it, let's differ it to the next round

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -21,11 +21,11 @@ jobs:
         uses: actions/cache@v2
         with:
           path: "**/node_modules"
-          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/package-lock.json') }}
       - name: Install dependencies
-        run: yarn install
+        run: npm install
```
[the remark this answers is not in the document]
Comment (danprinz) [reply .github/workflows/ci.yml:26]: reverted

**Review by nicolasbrugneaux — APPROVED** [review]

**Pull request merged**